### PR TITLE
Issue 1067: Problems with Partitioned Topics which name contains -partition-N

### LINKED ISSUE #1067: Problems with Partitioned Topics which name contains -partition-N
Someone accidentally created a partitioned topic in one of our cluster with a name which contains `-partition-2`.
This raised all sorts of issues, it would seem that only one partition was created for this topic, but metadata exists saying that it has 10 partitions.
When trying to get the subscriptions on the topic via HTTP the requests hang forever because we never merged https://github.com/apache/incubator-pulsar/pull/564
When trying to get partitioned-stats for the topic a warning is logged saying that no partition metadata was found.

We should probably:
1 - Don't allow partitioned topics which contain `-partition-` on its name
2 - Fix logic to allow those topics

### PR DESCRIPTION
**Pull request by sijie**
### Motivation

Fixes #1067.

Someone accidentally created a partitioned topic in one of our cluster with a name which contains -partition-2.
This raised all sorts of issues, it would seem that only one partition was created for this topic, but metadata exists saying that it has 10 partitions.

 ### Changes

Disallow creating a partitioned topic contains '-partition-' in the name.

### CODE REVIEW

**Comment by srkukarni** [comment]
rerun java tests

**Comment by sijie** [comment]
rerun java8 tests

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java [hunk 1]
```diff
@@ -254,6 +254,15 @@ protected void validateTopicName(String property, String namespace, String encod
         this.topicName = TopicName.get(domain(), namespaceName, topic);
     }
 
+    protected void validatePartitionedTopicName(String tenant, String namespace, String encodedTopic) {
+        // first, it has to be a validate topic name
+        validateTopicName(tenant, namespace, encodedTopic);
+        // second, "-partition-" is not allowed
+        if (encodedTopic.contains("-partition-")) {
```
Comment (rdhabalia) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java]: can we use `TopicName.PARTITIONED_TOPIC_SUFFIX` instead a plain string?

Comment (sijie) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/AdminResource.java]: changed it to use the constant

**Pull request merged**